app/controllers/application_controller.rb

```diff
@@ -1,5 +1,12 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    User.first
+  protect_from_forgery with: :exception
+
+  before_action :update_allowed_parameters, if: :devise_controller?
+
```

Comment:
- [ ] Users who are not signed can still visit different routes in your app. Kindly add the `authenticate_user!` helper to redirect the user to log in if they are not signed in.

  ```
  before_action :authenticate_user!
  ```